gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -0,0 +1,480 @@
+    assertThat(feature.getCaption()).isNull();
+    assertThat(point).isEqualTo(feature.getPoint());
+    assertThat(TEST_USER).isEqualTo(feature.getCreated().getUser());
+    assertThat(TEST_USER).isEqualTo(feature.getLastModified().getUser());
+  }
+
+  @Before
+  public void setUp() {
+    DaggerTestComponent.create().inject(this);
+  }
+
+  @Test
+  public void testInsertProject() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).test().assertComplete();
+  }
+
+  @Test
+  public void testGetProjects() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.getProjects().test().assertValue(ImmutableList.of(TEST_PROJECT));
+  }
+
+  @Test
+  public void testGetProjectById() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.getProjectById("project id").test().assertValue(TEST_PROJECT);
+  }
+
+  @Test
+  public void testDeleteProject() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.deleteProject(TEST_PROJECT).test().assertComplete();
+    localDataStore.getProjects().test().assertValue(AbstractCollection::isEmpty);
+  }
+
+  @Test
+  public void testInsertUser() {
+    localDataStore.insertOrUpdateUser(TEST_USER).test().assertComplete();
+  }
+
+  @Test
+  public void testGetUser() {
+    localDataStore.insertOrUpdateUser(TEST_USER).blockingAwait();
+    localDataStore.getUser("user id").test().assertValue(TEST_USER);
+  }
+
+  @Test
+  public void testApplyAndEnqueue_featureMutation() {
+    localDataStore.insertOrUpdateUser(TEST_USER).blockingAwait();
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+
+    localDataStore.applyAndEnqueue(TEST_FEATURE_MUTATION).test().assertComplete();
+
+    // assert that mutation is saved to local database
+    localDataStore
+        .getPendingMutations("feature id")
+        .test()
+        .assertValue(ImmutableList.of(TEST_FEATURE_MUTATION));
+
+    // assert feature is saved to local database
+    Feature feature = localDataStore.getFeature(TEST_PROJECT, "feature id").blockingGet();
+    assertFeature("feature id", TEST_POINT, feature);
+  }
+
+  @Test
+  public void testGetFeaturesOnceAndStream() {
+    localDataStore.insertOrUpdateUser(TEST_USER).blockingAwait();
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+
+    TestSubscriber<ImmutableSet<Feature>> subscriber =
+        localDataStore.getFeaturesOnceAndStream(TEST_PROJECT).test();
+
+    subscriber.assertValueCount(1);
+    subscriber.assertValueAt(0, AbstractCollection::isEmpty);
+
+    FeatureMutation mutation = TEST_FEATURE_MUTATION;
+    localDataStore.applyAndEnqueue(mutation).blockingAwait();
+
+    Feature feature =
+        localDataStore.getFeature(TEST_PROJECT, mutation.getFeatureId()).blockingGet();
+
+    subscriber.assertValueCount(2);
+    subscriber.assertValueAt(0, AbstractCollection::isEmpty);
```

Comment:
We could also use `subscriber.assertValues()` here instead of two `assertValueAt` calls.